src/index/txospenderindex.cpp

```diff
@@ -0,0 +1,60 @@
+#include <index/txospenderindex.h>
+
+#include <index/disktxpos.h>
+#include <node/blockstorage.h>
+#include <util/system.h>
+#include <validation.h>
+
+constexpr uint8_t DB_TXOSPENDERINDEX{'s'};
+
+std::unique_ptr<TxoSpenderIndex> g_txospenderindex;
+
+/** Access to the txo spender index database (indexes/txospenderindex/) */
+class TxoSpenderIndex::DB : public BaseIndex::DB
+{
+public:
+    explicit DB(size_t n_cache_size, bool f_memory = false, bool f_wipe = false);
+
+    bool WriteSpenderInfos(const std::vector<std::pair<COutPoint, uint256>>& items);
+};
+
+TxoSpenderIndex::DB::DB(size_t n_cache_size, bool f_memory, bool f_wipe) :
+    BaseIndex::DB(gArgs.GetDataDirNet() / "indexes" / "txospenderindex", n_cache_size, f_memory, f_wipe)
+{}
+
+TxoSpenderIndex::TxoSpenderIndex(size_t n_cache_size, bool f_memory, bool f_wipe)
+    : m_db(std::make_unique<TxoSpenderIndex::DB>(n_cache_size, f_memory, f_wipe))
+{}
+
+TxoSpenderIndex::~TxoSpenderIndex() {}
+
+bool TxoSpenderIndex::DB::WriteSpenderInfos(const std::vector<std::pair<COutPoint, uint256>>& items)
+{
+    CDBBatch batch(*this);
+    for (const auto& tuple : items) {
+        batch.Write(std::make_pair(DB_TXOSPENDERINDEX, tuple.first), tuple.second);
```

Comment:
Is there any reason to use `std::make_pair(DB_TXOSPENDERINDEX, tuple.first)`  as the key instead of just  `tuple.first` ?

`DB_TXOSPENDERINDEX` seems redundant.

Reply:
This is how all other indexes work even when there is just one possible key. Maybe because it makes it easier to add new record type later with a new key prefix ?